tests/cvc5-rules.smt2

```diff
@@ -0,0 +1,356 @@
+; Experimental rules for the internal cvc5 rules.
+
+(declare-sort Int 0)
+
+(declare-const = (-> (! Type :var T) T T Bool))
+
+(declare-const and (-> Bool Bool Bool))
+(declare-const or (-> Bool Bool Bool))
+(declare-const not (-> Bool Bool))
+(declare-const => (-> Bool Bool Bool))
+(declare-const false Bool)
+
+; ASSUME
+; SCOPE
+
+; SUBS
+(declare-rule subs ((T Type) (F Bool :list) (t1 T) (t2 T) (ids Int :list))
+    :premises (F)
+    :args (t1 ids)
+    :conclusion (= T t1 t2)
+    :require ((t2) (= T t2 (Subst t F ids)))
+)
+
+; REWRITE
+(declare-rule rewrite ((T Type) (t1 T) (t2 T) (idr Int))
+    :args (t1 idr)
+    :conclusion (= T t1 t2)
+    :require ((t2) (= T t2 (Rewriter idr t)))
+)
+
+; EVALUATE
+(declare-rule evaluate ((T Type) (t1 T) (t2 T))
+    :args (t1)
+    :conclusion (= T t1 t2)
+    :require ((t2) (Evaluate idr t))
+)
+
+; MACRO_SR_EQ_INTRO
+(declare-rule marco_sr_eq_intro ((T Type) (F Bool :list) (t1 T) (t2 T) (idsar Int :list))
+    :premises (F)
+    :args (t1 idsar) ; We have to merge the three arguments since they are all Int
+    :conclusion t2
+    :require (= T t2 (RwSubstSk t1 F idsar))
+)
+
+; MACRO_SR_PRED_INTRO
+(declare-rule marco_sr_eq_intro ((T Type) (F Bool :list) (R Bool) (idsar Int :list))
```

Comment:
`idsar` should not be `Int`. We should make this properly typed e.g. by introducing `(declare-sort MethodId 0)`.

We do need to decide on how to handle default arguments (or perhaps we should not allow them.